docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -0,0 +1,228 @@
+
+*Wait for the future to complete, then retrieve the state separately*:
+
+<CodeGroup>
+    ```python Prefect 2
+    @task
+    async def my_task():
+    	...
+	
+	
+    @flow
+    async def my_flow():
+    	future = await my_task.submit()
+	    # State is returned by `wait()`
+	    state = await future.wait()
+	    return state
+    ```
+
+    ```python Prefect 3
+    @task
+    async def my_task():
+    	...
+	
+	
+    @flow
+    async def my_flow():
+	    future = my_task.submit()
+	    # Wait for future completion  
+	    # before retrieving the state
+	    future.wait()
+	    state = future.state
+	    return state
+    ```
+</CodeGroup>
+
+If you do not first wait for the future to complete, the run may raise an 'AttributeError: coroutine object has no attribute get' exception. 
+
+In Prefect 3, `PrefectFutures` resulting from submitted task runs are not automatically awaited/resolved before exiting a flow, unless they’re a dependency of other flows or tasks, so *you must wait for a task’s `PrefectFuture` resolution before the end of a flow*:
+
+<CodeGroup>
+    ```python Prefect 2
+    @task
+    def my_task():
+	    ...
+	
+	
+    @flow
+    def my_flow():
+	    # The flow will not exit until 
+	    # these futures are resolved
+	    futures = my_task.map([1, 2, 3, 4])
+    ```
+
+    ```python Prefect 3
+    @task
+    def my_task():
+	    ...
+	
+	
+    @flow
+    def my_flow():
+	    futures = my_task.map([1, 2, 3, 4])
+	    # You must wait for futures before exiting flow.
+	    futures.wait()
+    ```
+</CodeGroup>
+
+If you do not wait for the futures, the tasks may fail with a 'Flow must be running to start a task' error. 
+The one exception to this case is that `PrefectFuture`s returned from a flow will resolve to states before completing the flow.
+
+## Result persistence and caching enabled by default
+
+Prefect 3 supports transactional semantics - tasks runs can be grouped into larger transactions, enabling more resilient, idempotent workflows. 
+To support this, Prefect 3 has more eager, intelligent caching defaults than Prefect 2.
+
+Whenever a task runs, Prefect 3 first computes its cache key, then uses that cache key to evaluate whether that same task has run with the same inputs and as part of the same flow before. 
+If it has, the cached result from the previous run is used.
+If it hasn't, the task runs. 
+This means that tasks that may have re-run with the same inputs in Prefect 2 will not re-run in Prefect 3 unless the cache key is modified to be more precise.
+The cache key can be configured through the new global `cache_policy` setting, or through the `cache_key_fn` on a particular task.
+
+Results and cache keys are not stored with your Prefect Cloud account or Prefect server instance. 
+They are persisted entirely in your configured result location. 
+Result persistence, and the caching behavior it enables, can be disabled through the `persist_result=False` flag.
```

Comment:
I think we also need to call out that any time you use result persistence and want to run a task multiple times within the same flow you must explicitly use a NONE cache policy

```
from prefect.cache_policies import NONE

@task(cache_policy=NONE)
```